components/rate/demo/character-function.tsx

```diff
@@ -12,9 +12,9 @@ const customIcons: Record<number, React.ReactNode> = {
 
 const App: React.FC = () => (
   <>
-    <Rate defaultValue={2} character={({ index }: { index: number }) => index + 1} />
+    <Rate defaultValue={2} character={({ index }) => (index || 0) + 1} />
```

Comment:
感觉用类型保护去解 ts 报错更好一点？

```suggestion
    <Rate
      defaultValue={2}
      character={({ index }) => typeof index === 'number' ? index + 1 : 1}
    />
```